Fig is deprecated, use docker-compose instead

metal3d opened the pull request:
Fig has been replaced by Docker Compose, and is now deprecated. The new documentation is on the docker website:
http://docs.docker.com/compose/

The default name is now docker-compose.yml

Keats commented:
Could you update the readme as well please?

metal3d commented:
Actually, I think that's not suffisant... I tried to launch it and there are a lot of errors (package needed, server won't start). I will try to fixup the entire problems and to complete my pull-request. I will fix README file too ;)

Keats commented:
I actually stopped using fig for frontend stuff since it was a bit of a pain personally, I still use it for the backend though.
Did you have success with it?

metal3d commented:
Actually the main problem is not fig or docker-compose but Dockerfile and package.json that are not complete (AFAIK).

At first, Docker container need git and bzip2 to be able to get some packages and extract them (I fixed this) then the packaga.json needs at least gulp-karma that is not in package.json.

Now, containers starts and says:

```
Attaching to ngboilerplate_angular_1
angular_1 | [09:23:18] Using gulpfile /home/angular/gulpfile.js
angular_1 | 
angular_1 | /usr/local/lib/node_modules/gulp/bin/gulp.js:129
angular_1 |     gulpInst.start.apply(gulpInst, toRun);
angular_1 |                    ^
angular_1 | TypeError: Cannot call method 'apply' of undefined
angular_1 |     at /usr/local/lib/node_modules/gulp/bin/gulp.js:129:20
angular_1 |     at process._tickDomainCallback (node.js:492:13)
ngboilerplate_angular_1 exited with code 8
```

metal3d commented:
I think it's correct. 
gulpe hasn't "parallel" and "series" methods, that should be in an earlier version but that's disapeard.
karma asks to not use andy "gulp" plugin, so I removed the call to "gulp-karma" and use "karma" as is.
PhantomJS needs to be able to use freetype and fontconfig, I added them in Dockerfile as for bzip2 and git command that should be installed.

Finally, I added a port mapping to 3001 (not very mandatory but that's easier to not use the container ip to access it)

I fixed README file too.

You probably want to fix my english because I'm french and I'm sure I make a lot of mistakes ;)

Keats commented:
Actually .serie and .parallel are part of gulp 4 (not released yet but this project is using it).
I'll have a look in a bit.

PS: pas de problemes

metal3d commented:
Ok, that's what I thought, but I didn't found any 4 tag in the gulp repository, that's why I used the 3.8 version. I can try to get back v4 and retest.

PS: french :) ?

Keats commented:
Ouais.
Keeping it in english for non-french users though !
You can read how i upgraded the project on this post https://blog.wearewizards.io/migrating-to-gulp-4-by-example

I'll have a look at this project right now

metal3d commented:
No problem for english, I continued to speak in english exactly for the reason you give ;)

I'm trying the 4.0 right now and fixing my Dockerfile (to use gulp-cli 4 too). So please wait my next rebase/push. 

BTW: nice boilerplate

Keats commented:
I fixed all the gulp related bugs + updated packages, you should only need to to update the dockerfile and the readme

metal3d commented:
Damned I fixed too...

metal3d commented:
Done,
Docker container needs python, git, g++ and so on to not crash at build time.
With your changes and the fix of this PR, it seems to work as expected without any error (on docker-compose build then docker-compose up)

metal3d commented:
... it seems that I've got changes from your gulpfile...

Keats commented:
Great thanks !
